### PR TITLE
Add TypeDoc and re-organize Account API docs

### LINKED ISSUE #30: Auto-generated documentation with TSDocs
Documentation of this library should be auto-generated with `TypeDoc`, this PR https://github.com/ethereumjs/ethereumjs-util/pull/170 can serve as a reference (see also [technical docs on documentation](https://ethereumjs.readthedocs.io/en/latest/technical.html#documentation)).

For this the manual documentation comments in the `README` (https://github.com/ethereumjs/ethereumjs-account) should be integrated in the code along the way. The code examples should either kept integrated or separated, whichever makes more sense respectively is more practical.

Depends on https://github.com/ethereumjs/ethereumjs-account/pull/27

### PR DESCRIPTION
**Pull request by chikeichan**
Fixes #30

### CODE REVIEW

**Comment by s1na** [comment]
You're on :fire: :smile: one small comment: Indentation of the comments seems off from `serialize` onwards.

**Comment by holgerd77** [comment]
@chikeichan Yes, that's actually a nice one, cool! 😄

File: src/index.ts [hunk 1]
```diff
@@ -50,14 +92,29 @@ export default class Account {
     ethUtil.defineProperties(this, fields, data)
   }
 
+  /**
+   * Returns the RLP serialization of the account as a `Buffer`.
+   *
+   * @return {Buffer}
```
Comment (s1na) [inline src/index.ts]: If the method is annotated, the docs don't need to specify the type. Same comment for `isContract` and `setCode`.

File: src/index.ts [hunk 2]
```diff
@@ -96,6 +227,11 @@ export default class Account {
     })
   }
 
+  /**
+   * Returns a `Boolean` determining if the account is empty.
+   *
+   * @return {boolean} if account is empty
```
Comment (s1na) [inline src/index.ts]: Could you please add `boolean` instead to the method's annotation? `isEmpty(): boolean`

**Review by s1na — DISMISSED** [review]
Two small comments, otherwise looks good!

**Comment by holgerd77** [comment]
Hi @chikeichan, can you address the comments from @s1na so that we can merge here?

File: src/index.ts (hunk 2 above)
Comment (chikeichan) [reply src/index.ts]: updated!

File: src/index.ts (hunk 1 above)
Comment (chikeichan) [reply src/index.ts]: updated!

**Review by holgerd77 — APPROVED** [review]
Cool, thanks 👍, looks good now!

**Pull request merged**